### ISSUE TITLE
order(list_column) gives strange error

### ISSUE THREAD

**Issue by MichaelChirico** [Rdatatable/data.table#4214]
```
DT = data.table(
  V = list(71L, c(71L, 132L), c(71L, 132L, 155L), 97L, 155L)
)
DT[order(V)]
# Error: Column 2 is length 2 which differs from length of column 1 (1)
```

I had forgotten `V` is a list column (and not `character` which can be sorted), but it seems the command went further than expected.

**Comment by jangorecki** [comment]
The problem is slightly more complex, because `order` redirects to `forder` and this one actually accepts data.table/list objects as well. Thus your example simply translates to
```r
V = list(71L, c(71L, 132L), c(71L, 132L, 155L), 97L, 155L)
data.table:::forder(V)
```

**Comment by jangorecki** [comment]
related thread: https://stat.ethz.ch/pipermail/r-devel/2020-May/079500.html

**Comment by mattdowle** [comment]
Closed by #4343 just merged I think but the PR didn't use the word "closes" so it didn't update the status of this issue to closed automatically.   If it didn't really close this issue, just reopen please.
